### ISSUE TITLE
Building any sketch gives "error: 'TCC1_CH4' was not declared in this scope"

### ISSUE THREAD

**Issue by robrussell** [robotics-masters/mm1-hat-arduino#11]
Building either an empty sketch or the blinky example from the Arduino IDE gives a build error in the Robohat MM1 libraries.

Here's the full error output.

```
Arduino: 1.8.11 (Windows 10), Board: "Robo HAT MM1 (SAMD51), Enabled, 120 MHz (standard), Small (-Os) (standard), Arduino, On"

C:\Program Files (x86)\Arduino\arduino-builder -dump-prefs -logger=machine -hardware C:\Program Files (x86)\Arduino\hardware -hardware C:\Users\rrssll\AppData\Local\Arduino15\packages -hardware C:\Users\rrssll\Documents\Arduino\hardware -tools C:\Program Files (x86)\Arduino\tools-builder -tools C:\Program Files (x86)\Arduino\hardware\tools\avr -tools C:\Users\rrssll\AppData\Local\Arduino15\packages -built-in-libraries C:\Program Files (x86)\Arduino\libraries -libraries C:\Users\rrssll\Documents\Arduino\libraries -fqbn=roboticsmasters:samd:robo_hat_mm1_m4:cache=on,speed=120,opt=small,usbstack=arduino,debug=on -ide-version=10811 -build-path C:\Users\rrssll\AppData\Local\Temp\arduino_build_666093 -warnings=none -build-cache C:\Users\rrssll\AppData\Local\Temp\arduino_cache_559680 -prefs=build.warn_data_percentage=75 -prefs=runtime.tools.arm-none-eabi-gcc.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\arm-none-eabi-gcc\7-2017q4 -prefs=runtime.tools.arm-none-eabi-gcc-7-2017q4.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\arm-none-eabi-gcc\7-2017q4 -prefs=runtime.tools.CMSIS-Atmel.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\CMSIS-Atmel\1.2.0 -prefs=runtime.tools.CMSIS-Atmel-1.2.0.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\CMSIS-Atmel\1.2.0 -prefs=runtime.tools.openocd.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\openocd\0.10.0-arduino9 -prefs=runtime.tools.openocd-0.10.0-arduino9.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\openocd\0.10.0-arduino9 -prefs=runtime.tools.CMSIS.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\CMSIS\4.5.0 -prefs=runtime.tools.CMSIS-4.5.0.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\CMSIS\4.5.0 -prefs=runtime.tools.bossac.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\bossac\1.8.0-48-gb176eee -prefs=runtime.tools.bossac-1.8.0-48-gb176eee.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\bossac\1.8.0-48-gb176eee -verbose C:\Users\rrssll\AppData\Local\Temp\untitled375011728.tmp\sketch_feb01a\sketch_feb01a.ino
C:\Program Files (x86)\Arduino\arduino-builder -compile -logger=machine -hardware C:\Program Files (x86)\Arduino\hardware -hardware C:\Users\rrssll\AppData\Local\Arduino15\packages -hardware C:\Users\rrssll\Documents\Arduino\hardware -tools C:\Program Files (x86)\Arduino\tools-builder -tools C:\Program Files (x86)\Arduino\hardware\tools\avr -tools C:\Users\rrssll\AppData\Local\Arduino15\packages -built-in-libraries C:\Program Files (x86)\Arduino\libraries -libraries C:\Users\rrssll\Documents\Arduino\libraries -fqbn=roboticsmasters:samd:robo_hat_mm1_m4:cache=on,speed=120,opt=small,usbstack=arduino,debug=on -ide-version=10811 -build-path C:\Users\rrssll\AppData\Local\Temp\arduino_build_666093 -warnings=none -build-cache C:\Users\rrssll\AppData\Local\Temp\arduino_cache_559680 -prefs=build.warn_data_percentage=75 -prefs=runtime.tools.arm-none-eabi-gcc.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\arm-none-eabi-gcc\7-2017q4 -prefs=runtime.tools.arm-none-eabi-gcc-7-2017q4.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\arm-none-eabi-gcc\7-2017q4 -prefs=runtime.tools.CMSIS-Atmel.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\CMSIS-Atmel\1.2.0 -prefs=runtime.tools.CMSIS-Atmel-1.2.0.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\CMSIS-Atmel\1.2.0 -prefs=runtime.tools.openocd.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\openocd\0.10.0-arduino9 -prefs=runtime.tools.openocd-0.10.0-arduino9.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\openocd\0.10.0-arduino9 -prefs=runtime.tools.CMSIS.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\CMSIS\4.5.0 -prefs=runtime.tools.CMSIS-4.5.0.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\CMSIS\4.5.0 -prefs=runtime.tools.bossac.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\bossac\1.8.0-48-gb176eee -prefs=runtime.tools.bossac-1.8.0-48-gb176eee.path=C:\Users\rrssll\AppData\Local\Arduino15\packages\arduino\tools\bossac\1.8.0-48-gb176eee -verbose C:\Users\rrssll\AppData\Local\Temp\untitled375011728.tmp\sketch_feb01a\sketch_feb01a.ino
Using board 'robo_hat_mm1_m4' from platform in folder: C:\Users\rrssll\AppData\Local\Arduino15\packages\roboticsmasters\hardware\samd\0.0.24
Using core 'arduino' from platform in folder: C:\Users\rrssll\AppData\Local\Arduino15\packages\roboticsmasters\hardware\samd\0.0.24
Detecting libraries used...
"C:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\arm-none-eabi-gcc\\7-2017q4/bin/arm-none-eabi-g++" -mcpu=cortex-m4 -mthumb -c -g -Os -w -std=gnu++11 -ffunction-sections -fdata-sections -fno-threadsafe-statics -nostdlib --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -w -x c++ -E -CC -DF_CPU=120000000L -DARDUINO=10811 -DARDUINO_ROBOHATMM1_M4 -DARDUINO_ARCH_SAMD -D__SAMD51G19A__ -DROBOTICSMASTERS_ROBOHATMM1_M4 -D__SAMD51__ -DUSB_VID=0x1209 -DUSB_PID=0x4D45 -DUSBCON "-DUSB_MANUFACTURER=\"Robotics Masters LTD\"" "-DUSB_PRODUCT=\"Robo HAT MM1 M4\"" -g "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\cores\\arduino/Adafruit_TinyUSB_Core" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\cores\\arduino/Adafruit_TinyUSB_Core/tinyusb/src" -D__FPU_PRESENT -DARM_MATH_CM4 -DCRYSTALLESS -mfloat-abi=hard -mfpu=fpv4-sp-d16 -DENABLE_CACHE "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\CMSIS\\4.5.0/CMSIS/Include/" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\CMSIS-Atmel\\1.2.0/CMSIS/Device/ATMEL/" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\cores\\arduino" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\variants\\robohatmm1_m4" "C:\\Users\\rrssll\\AppData\\Local\\Temp\\arduino_build_666093\\sketch\\sketch_feb01a.ino.cpp" -o nul
Generating function prototypes...
"C:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\arm-none-eabi-gcc\\7-2017q4/bin/arm-none-eabi-g++" -mcpu=cortex-m4 -mthumb -c -g -Os -w -std=gnu++11 -ffunction-sections -fdata-sections -fno-threadsafe-statics -nostdlib --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -w -x c++ -E -CC -DF_CPU=120000000L -DARDUINO=10811 -DARDUINO_ROBOHATMM1_M4 -DARDUINO_ARCH_SAMD -D__SAMD51G19A__ -DROBOTICSMASTERS_ROBOHATMM1_M4 -D__SAMD51__ -DUSB_VID=0x1209 -DUSB_PID=0x4D45 -DUSBCON "-DUSB_MANUFACTURER=\"Robotics Masters LTD\"" "-DUSB_PRODUCT=\"Robo HAT MM1 M4\"" -g "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\cores\\arduino/Adafruit_TinyUSB_Core" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\cores\\arduino/Adafruit_TinyUSB_Core/tinyusb/src" -D__FPU_PRESENT -DARM_MATH_CM4 -DCRYSTALLESS -mfloat-abi=hard -mfpu=fpv4-sp-d16 -DENABLE_CACHE "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\CMSIS\\4.5.0/CMSIS/Include/" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\CMSIS-Atmel\\1.2.0/CMSIS/Device/ATMEL/" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\cores\\arduino" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\variants\\robohatmm1_m4" "C:\\Users\\rrssll\\AppData\\Local\\Temp\\arduino_build_666093\\sketch\\sketch_feb01a.ino.cpp" -o "C:\\Users\\rrssll\\AppData\\Local\\Temp\\arduino_build_666093\\preproc\\ctags_target_for_gcc_minus_e.cpp"
"C:\\Program Files (x86)\\Arduino\\tools-builder\\ctags\\5.8-arduino11/ctags" -u --language-force=c++ -f - --c++-kinds=svpf --fields=KSTtzns --line-directives "C:\\Users\\rrssll\\AppData\\Local\\Temp\\arduino_build_666093\\preproc\\ctags_target_for_gcc_minus_e.cpp"
Compiling sketch...
"C:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\arm-none-eabi-gcc\\7-2017q4/bin/arm-none-eabi-g++" -mcpu=cortex-m4 -mthumb -c -g -Os -w -std=gnu++11 -ffunction-sections -fdata-sections -fno-threadsafe-statics -nostdlib --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -MMD -DF_CPU=120000000L -DARDUINO=10811 -DARDUINO_ROBOHATMM1_M4 -DARDUINO_ARCH_SAMD -D__SAMD51G19A__ -DROBOTICSMASTERS_ROBOHATMM1_M4 -D__SAMD51__ -DUSB_VID=0x1209 -DUSB_PID=0x4D45 -DUSBCON "-DUSB_MANUFACTURER=\"Robotics Masters LTD\"" "-DUSB_PRODUCT=\"Robo HAT MM1 M4\"" -g "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\cores\\arduino/Adafruit_TinyUSB_Core" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\cores\\arduino/Adafruit_TinyUSB_Core/tinyusb/src" -D__FPU_PRESENT -DARM_MATH_CM4 -DCRYSTALLESS -mfloat-abi=hard -mfpu=fpv4-sp-d16 -DENABLE_CACHE "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\CMSIS\\4.5.0/CMSIS/Include/" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\CMSIS-Atmel\\1.2.0/CMSIS/Device/ATMEL/" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\cores\\arduino" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\variants\\robohatmm1_m4" "C:\\Users\\rrssll\\AppData\\Local\\Temp\\arduino_build_666093\\sketch\\sketch_feb01a.ino.cpp" -o "C:\\Users\\rrssll\\AppData\\Local\\Temp\\arduino_build_666093\\sketch\\sketch_feb01a.ino.cpp.o"
Compiling libraries...
Compiling core...
"C:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\arm-none-eabi-gcc\\7-2017q4/bin/arm-none-eabi-g++" -mcpu=cortex-m4 -mthumb -c -g -Os -w -std=gnu++11 -ffunction-sections -fdata-sections -fno-threadsafe-statics -nostdlib --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -MMD -DF_CPU=120000000L -DARDUINO=10811 -DARDUINO_ROBOHATMM1_M4 -DARDUINO_ARCH_SAMD -D__SAMD51G19A__ -DROBOTICSMASTERS_ROBOHATMM1_M4 -D__SAMD51__ -DUSB_VID=0x1209 -DUSB_PID=0x4D45 -DUSBCON "-DUSB_MANUFACTURER=\"Robotics Masters LTD\"" "-DUSB_PRODUCT=\"Robo HAT MM1 M4\"" -g "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\cores\\arduino/Adafruit_TinyUSB_Core" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\cores\\arduino/Adafruit_TinyUSB_Core/tinyusb/src" -D__FPU_PRESENT -DARM_MATH_CM4 -DCRYSTALLESS -mfloat-abi=hard -mfpu=fpv4-sp-d16 -DENABLE_CACHE "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\CMSIS\\4.5.0/CMSIS/Include/" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\arduino\\tools\\CMSIS-Atmel\\1.2.0/CMSIS/Device/ATMEL/" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\cores\\arduino" "-IC:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\variants\\robohatmm1_m4" "C:\\Users\\rrssll\\AppData\\Local\\Arduino15\\packages\\roboticsmasters\\hardware\\samd\\0.0.24\\variants\\robohatmm1_m4\\variant.cpp" -o "C:\\Users\\rrssll\\AppData\\Local\\Temp\\arduino_build_666093\\core\\variant.cpp.o"
C:\Users\rrssll\AppData\Local\Arduino15\packages\roboticsmasters\hardware\samd\0.0.24\variants\robohatmm1_m4\variant.cpp:39:62: error: 'TCC1_CH4' was not declared in this scope

   { PORTA,  20, PIO_DIGITAL, PIN_ATTR_PWM_F, No_ADC_Channel, TCC1_CH4, NOT_ON_TIMER, EXTERNAL_INT_4 }, //

                                                              ^~~~~~~~

C:\Users\rrssll\AppData\Local\Arduino15\packages\roboticsmasters\hardware\samd\0.0.24\variants\robohatmm1_m4\variant.cpp:39:62: note: suggested alternative: 'TCC1_CH3'

   { PORTA,  20, PIO_DIGITAL, PIN_ATTR_PWM_F, No_ADC_Channel, TCC1_CH4, NOT_ON_TIMER, EXTERNAL_INT_4 }, //

                                                              ^~~~~~~~

                                                              TCC1_CH3

C:\Users\rrssll\AppData\Local\Arduino15\packages\roboticsmasters\hardware\samd\0.0.24\variants\robohatmm1_m4\variant.cpp:40:62: error: 'TCC1_CH5' was not declared in this scope

   { PORTA,  21, PIO_DIGITAL, PIN_ATTR_PWM_F, No_ADC_Channel, TCC1_CH5, NOT_ON_TIMER, EXTERNAL_INT_5 }, //

                                                              ^~~~~~~~

C:\Users\rrssll\AppData\Local\Arduino15\packages\roboticsmasters\hardware\samd\0.0.24\variants\robohatmm1_m4\variant.cpp:40:62: note: suggested alternative: 'TCC1_CH3'

   { PORTA,  21, PIO_DIGITAL, PIN_ATTR_PWM_F, No_ADC_Channel, TCC1_CH5, NOT_ON_TIMER, EXTERNAL_INT_5 }, //

                                                              ^~~~~~~~

                                                              TCC1_CH3

exit status 1
Error compiling for board Robo HAT MM1 (SAMD51).
```

**Comment by wallarug** [comment]
Hey @robrussell ,

I'll take a look at this one.  TCC1_CH4 does not exist on the SAMD51.  There may need to be some adjustments in the Arduino Code.

**Comment by wallarug** [comment]
Hi @robrussell .

This should be fixed by #12 now.

**Comment by robrussell** [comment]
That did it, thanks!
